### PR TITLE
Issue #757: Renaming USS directory changes tree structure

### PR DESCRIPTION
**Pull request by katelynienaber**
## Proposed changes

This is a fix for [issue #757](https://github.com/zowe/vscode-extension-for-zowe/issues/757). I added an error message so that if a user enters a new name for a folder, and a folder with that name already exists, then the user will get an error message and have to choose a different name.

@zFernand0 I posted a question about your comment on the original issue, please, if you have time please explain what you meant ^^'

folderRename

## Release Notes
Milestone: 1.9
Changelog: Added error message for USS folder renaming when the new name is already used.

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [x] gif or screenshot is included if visual changes are made
- [x] `npm run vscode:prepublish` has been executed
- [x] All checks have passed (DCO, Jenkins and Code Coverage)
- [x] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [x] There is coverage for the code that I have added
- [x] I have tested it manually and there are no regressions found
- [x] I have added necessary documentation (if appropriate)
- [x] Any PR dependencies have been merged and published (if appropriate)

### CODE REVIEW

**Comment by jellypuno** [comment]
Is this an exemption?

RenameUSSDir2

**Comment by katelynienaber** [comment]
> Is this an exemption?

Fixed this :) Yes it was, and no it shouldn't have been...

**Comment by jellypuno** [comment]
I think this codecov warnings are new:

image

**Comment by katelynienaber** [comment]
> I think this codecov warnings are new:
> 
> image

Function `checkDuplicateLabel` is already covered in tests...see line [249](https://github.com/zowe/vscode-extension-for-zowe/blob/f8b3d43b009e889b18e734553a324eccc9898d61/__tests__/__unit__/uss/USSTree.unit.test.ts#L249)

**Review by jellypuno — APPROVED** [review]

**Comment by katelynienaber** [comment]
> Functionality works great for me and all tests are passing.
> 
> I also tested this in Theia and I can confirm that this PR also addresses issue 755. The eTag error does not appear anymore.

Hey, sorry to bother :) Could you approve the PR officially if you are satisfied that there are no bugs?

**Review by crawr — APPROVED** [review]
LGTM

**Comment by lauren-li** [comment]
All unit and expected integration tests are passing and the new functionalities in this PR work well, but I think the issue might also happen when renaming a file to have the same name as an existing folder at the same level:
ze-ussRename

Although the gif shows a single "file" item with the folder's name at the end, if I refresh a couple more times, it eventually shows up as a single folder of the same name containing the un-renamed file (which is what the actual end-result is, based on when I double-checked via SSH):
Screen Shot 2021-01-12 at 6 40 27 AM

Do we want to apply a name-checking logic between files and folders, as well?

Thanks @katelynienaber for working on this!

~
(Also, this PR doesn't contain the `getChildren` looping issue I had mentioned being concerned about in scrum last week. I think I was remembering a different implementation of this fix at the time.)

**Comment by katelynienaber** [comment]
@lauren-li Hey, thanks for your review! :D

It should work now for folder & file names, compared to each other :) Please check again when you have time ^^

**Comment by lauren-li** [comment]
@katelynienaber Thanks for the updates! I will give it a try.

Could you also please move any changes from `__mocks__/**` and `i18n/**` into the `packages/zowe-explorer` folder? I just noticed it seems the PR has some extra folders hanging around at the top level (probably from the monorepo migration).

File: packages/zowe-explorer/src/uss/USSTree.ts [hunk 1]
```diff
@@ -151,6 +128,21 @@ export class USSTree extends ZoweTreeProvider implements IZoweTree<IZoweUSSTreeN
...
         }
     }
+
+    public checkDuplicateLabel(newFullPath: string, nodesToCheck: IZoweUSSTreeNode[], newNodeType: string) {
+        for (const node of nodesToCheck) {
+            const nodeType = contextually.isFolder(node) ? "folder" : "file";
+            if (newFullPath === node.fullPath.trim()) {
+                return localize(
+                    "renameUSSNode.duplicateName",
+                    "A {0} already exists with this name. Please choose a different one.",
```
Comment (lauren-li) [inline packages/zowe-explorer/src/uss/USSTree.ts:138]: @katelynienaber This is suuuper nitpicky, but could we please say, "Please choose a different name?"

I think it makes the instruction a bit clearer.

Comment (katelynienaber) [reply packages/zowe-explorer/src/uss/USSTree.ts]: I don't like reusing the word "name" twice in a row, but I guess clarity is more important than literary grace here XD

Fixed.

**Comment by katelynienaber** [comment]
> @katelynienaber Thanks for the updates! I will give it a try.
> 
> Could you also please move any changes from `__mocks__/**` and `i18n/**` into the `packages/zowe-explorer` folder? I just noticed it seems the PR has some extra folders hanging around at the top level (probably from the monorepo migration).

Yep, these have been removed :)

**Review by lauren-li — APPROVED** [review]
This looks good to me now. All unit and expected integration tests are passing. The newly-added functionality seems to work well, and adds nice UI feedback while reducing calls to the server. I did not find any regressions.

Thanks @katelynienaber for your hard work and patience on this PR!

**Review by crawr — APPROVED** [review]
Works wonderfully for me. I like how both file and directory node types are handled in this PR. Thanks @katelynienaber

**Pull request merged**